### PR TITLE
Add intra-site link instructions to docs

### PR DESCRIPTION
**Pull request by itsyme**
**What is the purpose of this pull request?**

- [x] Documentation update
- [ ] Bug fix
- [ ] Feature addition or enhancement
- [ ] Code maintenance
- [ ] DevOps
- [ ] Improve developer experience
- [ ] Others, please explain:

**Overview of changes:**
Related to #2273. Added a section in "Contribute to Documentation" to explain the use tags when adding intra-site links. Added a box to "Implement a New Feature" to remind developers to use tags in intra-site links.

**Anything you'd like to highlight/discuss:**

**Testing instructions:**

**Proposed commit message: (wrap lines at 72 characters)**
Add intra-site link instructions to docs

---

**Checklist:** :ballot_box_with_check:

- [ ] Updated the documentation for feature additions and enhancements
- [ ] Added tests for bug fixes or features
- [x] Linked all related issues
- [x] No unrelated changes

### CODE REVIEW

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 1]
```diff
@@ -91,6 +91,40 @@ Check that the following are true:
 
 </box>
 
+## Adding Intra-Site Links to Documentation
```
Comment (tlylt) [inline docs/devGuide/bootcamp/contributeToDocs.md:94]: Let's shift this into Workflow (create a Docs section in there), as the content included here is phrased in a BootCamp task manner.

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 2]
```diff
@@ -91,6 +91,40 @@ Check that the following are true:
...
+
+Due to the way MarkBind's documentation is deployed, links from the developer guide that refer to the user guide, or vice versa, have to be differentiated with tags. MarkBind's documentation currently has 3 tags:
```
Comment (tlylt) [inline docs/devGuide/bootcamp/contributeToDocs.md:96]: Can we add a note here that intra-site links within UG and DG need not add the tags

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 3]
```diff
@@ -91,6 +91,40 @@ Check that the following are true:
...
+
+- `environment--combined` for locally served documentation
```
Comment (tlylt) [inline docs/devGuide/bootcamp/contributeToDocs.md:98]: for deploying both the ug and dg tgt, I think that's the main point. It's not just locally served. It is also served on markbind-master, and during pr-preview

File: docs/devGuide/bootcamp/contributeToDocs.md [hunk 4]
```diff
@@ -91,6 +91,40 @@ Check that the following are true:
...
+- `environment--ug` for deployed User Guide
+- `environment--dg` for deployed Developer Guide
+
+Using tags ensures that the correct link is created in each environment.
+
```
Comment (tlylt) [inline docs/devGuide/bootcamp/contributeToDocs.md:103]: can we add a brief intro to say what the following is about? examples?

**Review by tlylt — CHANGES_REQUESTED** [review]

File: docs/devGuide/development/workflow.md [hunk 5]
```diff
@@ -258,6 +258,42 @@ Once you're done, be sure to run the `updatetest` script mentioned [above](#upda
 
 </div>
 
+## Documenting
+
+#### Adding Intra-Site Links to Documentation
```
Comment (tlylt) [inline docs/devGuide/development/workflow.md:263]:
```suggestion
### Adding Intra-Site Links to Documentation
```

File: docs/devGuide/development/workflow.md [hunk 6]
```diff
@@ -258,6 +258,42 @@ Once you're done, be sure to run the `updatetest` script mentioned [above](#upda
...
+
+Due to the way MarkBind's documentation is deployed, links from the developer guide that refer to the user guide, or vice versa, have to be differentiated with <a tags="environment--combined" href="/userGuide/tweakingThePageStructure.html#plugin-tags">tags</a><a tags="environment--dg" href="https://markbind.org/userGuide/tweakingThePageStructure.html#plugin-tags">tags</a>. Intra-site links within the developer guide or user guide need not have tags. MarkBind's documentation currently has 3 tags:
+
+- `environment--combined` for deploying the User Guide and Developer Guide together
+- `environment--ug` for deployed User Guide
+- `environment--dg` for deployed Developer Guide
+
+Tags allow you to selectively filter content when building a site. Using them with the links helps to filter the correct link to be rendered for the right environment, ensuring that the correct link is created.
+
+##### Developer Guide
```
Comment (tlylt) [inline docs/devGuide/development/workflow.md:273]:
```suggestion
#### Developer Guide
```

File: docs/devGuide/development/workflow.md [hunk 7]
```diff
@@ -258,6 +258,42 @@ Once you're done, be sure to run the `updatetest` script mentioned [above](#upda
...
+[Link Title](/userGuide/newPage.html)
+```
+
+%%REPLACED WITH:%%
+```html
+<a tags="environment--combined" href="/userGuide/newPage.html">Link Title</a>
+<a tags="environment--dg" href="https://markbind.org/userGuide/newPage.html">Link Title</a>
+```
+
+##### User Guide
```
Comment (tlylt) [inline docs/devGuide/development/workflow.md:285]:
```suggestion
#### User Guide
```

**Review by tlylt — CHANGES_REQUESTED** [review]
Can we include the rationale of why the tags work? (the part about using absolute link vs relative link)?

**Review by tlylt — APPROVED** [review]

File: docs/devGuide/development/workflow.md [hunk 8]
```diff
@@ -258,6 +258,42 @@ Once you're done, be sure to run the `updatetest` script mentioned [above](#upda
...
+
+- `environment--combined` for deploying the User Guide and Developer Guide together
+- `environment--ug` for deployed User Guide
+- `environment--dg` for deployed Developer Guide
+
+Tags allow you to selectively filter content when building a site. Using them with the links helps to filter the correct link to be rendered for the right environment, ensuring that the correct link is created. When the User Guide and Developer Guide are deployed together, both guides have access to the files from the other, allowing the relative link to work. However, when they are deployed individually, this is not the case. Using absolute links creates the same effect as the relative link without having to have access to the files from the other guide.
```
Comment (lhw-1) [inline docs/devGuide/development/workflow.md:271]:
```suggestion
To ensure that correct links are created, use tags with the links to selectively filter the correct link to be rendered for the right environment. When both the User Guide and the Developer Guide are deployed together, they have access to the files from each other, allowing the relative link to work. However, when they are deployed individually, this is not the case. Using absolute links creates the same effect as the relative link without having to have access to the files from the other guide.
```

**Review by lhw-1 — APPROVED** [review]
LGTM! Just some wording nits :)

**Pull request merged**